lib/beacon/router_server.ex

```diff
@@ -38,20 +38,11 @@ defmodule Beacon.RouterServer do
   # Client
 
   def lookup_page(site, path_info) when is_atom(site) and is_list(path_info) do
-    if Beacon.Config.fetch!(site).mode == :testing do
-      with {_path, page_id} <- lookup_path(site, path_info),
-           {:ok, page_module} <- Beacon.Loader.maybe_reload_page_module(site, page_id) do
-        Beacon.apply_mfa(page_module, :page, [])
-      else
-        _ -> nil
-      end
+    with {_path, page_id} <- lookup_path(site, path_info) do
+      page_module = Beacon.Loader.fetch_page_module(site, page_id)
+      Beacon.apply_mfa(page_module, :page, [])
     else
-      with {_path, page_id} <- lookup_path(site, path_info) do
-        page_module = Beacon.Loader.fetch_page_module(site, page_id)
-        Beacon.apply_mfa(page_module, :page, [])
-      else
-        _ -> nil
-      end
+      _ -> nil
     end
```

Comment:
this is now unified so all modes will use the error handler for loading (via apply/3)